homeworks/rostyslav.vyshemirskyi_Rooosss/PokemonProject/scripts/script.js

```diff
@@ -0,0 +1,86 @@
+const EMAIL_INPUT_ID = 'email';
+const PASSWORD_INPUT_ID = 'password';
+const NOT_A_ROBOT_CHECKBOX_ID = 'check';
+const SUBMIT_BUTTON_ID = 'submit-button';
+const ERRORS_CONTAINER_ID = 'errors-container';
+const RESULT_PAGE_PATH = './success.html';
+
+const submitButton = document.getElementById(SUBMIT_BUTTON_ID);
+
+/**
+ * Return input value by id.
+ * @param {string} elementId
+ * @return {string|boolean} input value
+ */
+function getValueById(elementId) {
+    const element = document.getElementById(elementId);
+    const type = element.getAttribute('type');
+    return type === 'checkbox' ? element.checked : element.value;
+}
+/**
+ * Add errors to errors container.
+ * @param {Object} inputData in format like: { [input_id]: error_text, ... }
+ */
+function setErrors(inputData) {
+    const errorContainerElement = document.getElementById(ERRORS_CONTAINER_ID);
+    Object.values(inputData).forEach((error) => {
+        const errorElement = document.createElement('p');
+        errorElement.classList.add('error');
+        errorElement.textContent = error;
+        errorContainerElement.appendChild(errorElement);
+    });
+}
+
+/**
+ * Delete all errors from errors container.
+ */
+function deleteErrors() {
+    const errorContainerElement = document.getElementById(ERRORS_CONTAINER_ID);
+    errorContainerElement.replaceChildren();
+}
+
+/**
+ * Goes to the page with the result.
+ */
+function navigateToResultPage() {
+    window.location.href = RESULT_PAGE_PATH;
+}
+
+function isEmail(email) {
+    return /\S+@\S+\.\S+/.test(email);
+}
+
+function deleteSpaces(str) {
+    return str.replace(/\s/g, '');
+}
+
+function validateForm() {
+    const email = deleteSpaces(getValueById(EMAIL_INPUT_ID));
+    const password = getValueById(PASSWORD_INPUT_ID);
+    const check = getValueById(NOT_A_ROBOT_CHECKBOX_ID);
+    deleteErrors();
+
+    if (!email) {
+        setErrors({ [EMAIL_INPUT_ID]: 'the email field must be filled' });
+        return false;
+    } if (!isEmail(email)) {
+        setErrors({ [EMAIL_INPUT_ID]: 'email is not valid' });
+        return false;
+    }
+
+    if (!password) {
+        setErrors({ [PASSWORD_INPUT_ID]: 'the password field must be filled' });
+        return false;
+    } if (password.length < 8 || password.length > 12) {
+        setErrors({ [PASSWORD_INPUT_ID]: 'password is very short or long' });
+        return false;
+    }
+
+    if (!check) {
+        setErrors({ [NOT_A_ROBOT_CHECKBOX_ID]: 'please check this element' });
```

Comment:
Which element should I check? Proly, the better message would be: `Please check not a robot checkbox` or `Please confirm, that you are not a robot`
![image](https://github.com/MastersAcademy/fe-course-basic-2023/assets/98031768/ac8a16eb-ea3a-4278-a305-967ad66a31d6)